docs/resources/domain.md

```diff
@@ -36,6 +36,7 @@ The following arguments are supported:
     the domain will accept email for sub-domains.
 * `dkim_key_size` - (Optional) The length of your domain’s generated DKIM key. Default value is `1024`.
 * `dkim_selector` - (Optional) The name of your DKIM selector if you want to specify it whereas MailGun will make it's own choice.
+* `force_dkim_authority` - (Optional) If set to true, the domain will be the DKIM authority for itself even if the root domain is registered on the same mailgun account. If set to false, the domain will have the same DKIM authority as the root domain registered on the same mailgun account
```

Comment:
Could you provide info what is the default value?

Reply:
It could be nice to add these fields to unit test. After all let me know if `make testacc` works for you.

Reply:
Default value added!

I got errors when run `make testacc`, even on master:
```
$ MAILGUN_API_KEY=XXX make testacc

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v  -timeout 120m -count=1
?   	github.com/terraform-providers/terraform-provider-mailgun	[no test files]
=== RUN   TestAccMailgunDomainDataSource_Basic
2022/09/29 16:28:49 [DEBUG] resourceMailgunCredential()
2022/09/29 16:28:49 [DEBUG] resourceMailgunWebhook()
    data_source_mailgun_domain_test.go:15: Step 1/1 error: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
    testing_new.go:63: Error retrieving state, there may be dangling resources: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
--- FAIL: TestAccMailgunDomainDataSource_Basic (0.26s)
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccMailgunDomainCredential_Basic
    resource_mailgun_credential_test.go:20: Step 1/1 error: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
    testing_new.go:63: Error retrieving state, there may be dangling resources: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
--- FAIL: TestAccMailgunDomainCredential_Basic (0.12s)
=== RUN   TestAccMailgunDomainCredential_Update
    resource_mailgun_credential_test.go:50: Step 1/2 error: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
    testing_new.go:63: Error retrieving state, there may be dangling resources: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
--- FAIL: TestAccMailgunDomainCredential_Update (0.12s)
=== RUN   TestAccMailgunDomain_Basic
    resource_mailgun_domain_test.go:19: Step 1/1 error: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
    testing_new.go:63: Error retrieving state, there may be dangling resources: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
--- FAIL: TestAccMailgunDomain_Basic (0.12s)
=== RUN   TestAccMailgunDomain_Import
    resource_mailgun_domain_test.go:52: Step 1/2 error: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
    testing_new.go:63: Error retrieving state, there may be dangling resources: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
--- FAIL: TestAccMailgunDomain_Import (0.12s)
=== RUN   TestAccMailgunRoute_Basic
    resource_mailgun_route_test.go:17: Step 1/1 error: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
    testing_new.go:63: Error retrieving state, there may be dangling resources: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
--- FAIL: TestAccMailgunRoute_Basic (0.12s)
=== RUN   TestAccMailgunRoute_Import
    resource_mailgun_route_test.go:44: Step 1/2 error: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
    testing_new.go:63: Error retrieving state, there may be dangling resources: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
--- FAIL: TestAccMailgunRoute_Import (0.11s)
=== RUN   TestAccMailgunWebhook_Basic
    resource_mailgun_webhook_test.go:19: Step 1/1 error: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
    testing_new.go:63: Error retrieving state, there may be dangling resources: unsupported state format version: expected ["0.1" "0.2"], got "1.0"
--- FAIL: TestAccMailgunWebhook_Basic (0.11s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-mailgun/mailgun	1.989s
FAIL
make: *** [testacc] Error 1
```

I confess that I have no idea what's going on ;) I'm fairly new to Go, Terraform etc...!
Note: I have created a new Mailgun account for this tests. Is there any requirement about the state of the MG account ?

It is not as relevant as a real unit test but... I was able to create/update MG domains in my TF project.

Reply:
Hm... yeah, I believe that it's related to TF incompatibility, try to bump terraform sdk version to latest: https://github.com/hashicorp/terraform-plugin-sdk/ in `go.mod`, completely remove all .terraform* files and  re-run the tests.